app/features/worksheet/tools.py

```diff
@@ -0,0 +1,278 @@
+        if '_' in getattr(validated_response, self.main_field):
+            return True
+        else:
+            if self.verbose:
+                logger.warning(f"{self.section_name} does not have blank ___.")
+            return False
+
+# Open ended question subclass
+class OpenEndedQuestion(QuestionBase):
+    def set_names(self):
+        self.section_name = 'Open-ended question'
+        self.main_field = 'question'
+    
+    def sub_validate(self):
+        OpenEndedQuestionFormat(**self.response)
+        return True
+
+# True False question subclass
+class TrueFalseQuestion(QuestionBase):
+    def set_names(self):
+        self.section_name = 'True or False question'
+        self.main_field = 'question'
+    
+    def sub_validate(self):
+        TrueFalseQuestionFormat(**self.response)
+        return True
+
+# Create worksheets
+class WorksheetBuilder:
+    def __init__(self, topic, grade_level, model=None, embedding_model = None, prompt_summary=None, parser_summary = None, prompt_fill_in_blank=None, parser_fill_in_blank=None, prompt_multiple_choice=None, parser_multiple_choice=None, prompt_open_ended = None, parser_open_ended = None, prompt_true_false = None, parser_true_false = None, verbose=False):
+        if topic is None or grade_level is None:
+            raise ValueError("Topic and Grade level must be provided")
+        
+        default_config = {
+            "model": GoogleGenerativeAI(model="gemini-1.0-pro", temperature=0.5),
+            "embedding_model": SentenceTransformer('all-MiniLM-L6-v2'),
+            "prompt_summary": read_text_file("prompts/worksheet_prompt_summary.txt"),
+            "parser_summary": JsonOutputParser(pydantic_object=SummaryFormat),
+            "prompt_fill_in_blank": read_text_file('prompts/worksheet_prompt_fill_in_blank.txt'),
+            "parser_fill_in_blank": JsonOutputParser(pydantic_object = FillinblankQuestionFormat),
+            "prompt_multiple_choice": read_text_file("prompts/worksheet_prompt_multiple_choice.txt"),
+            "parser_multiple_choice": JsonOutputParser(pydantic_object = MultipleChoiceQuestionFormat),
+            "prompt_open_ended": read_text_file("prompts/worksheet_prompt_open_ended.txt"),
+            "parser_open_ended": JsonOutputParser(pydantic_object = OpenEndedQuestionFormat),
+            "prompt_true_false": read_text_file("prompts/worksheet_prompt_true_false.txt"),
+            "parser_true_false": JsonOutputParser(pydantic_object = TrueFalseQuestionFormat)
+        }
+        
+        self.model = model or default_config["model"]
+        self.embedding_model = embedding_model or default_config["embedding_model"]
+        self.prompt_summary = prompt_summary or default_config["prompt_summary"]
+        self.parser_summary = parser_summary or default_config["parser_summary"]
+        self.prompt_fill_in_blank = prompt_fill_in_blank or default_config["prompt_fill_in_blank"]
```

Comment:
Use another method for reusing the components and having a better scalability. Don't generate variables for each value, reuse the previous ones.

Reply:
DRY